### DISCUSSION TITLE
Improve Clarity combobox

### DISCUSSION THREAD

**Discussion by Callen93** [vmware-clarity/ng-clarity#171] [Ideas]
Would it be possible to add a feature, allowing the addition of a custom value ?

**Comment by d-m-s** [comment]
I do have the same request, just wanted to add more details to it!

# Combobox: support for custom values
The combobox is a really helpful component, it just lacks the addition of custom values.
Whenever custom values are needed, currently the [datalist ](https://clarity.design/documentation/datalist)is used as a replacement, but this results in inconsistent user-interfaces. Datalist also has some display issues in different browsers.
In the following usecase the datalist can not serve as replacement.

## Usecase Tags/Labels

- We're using it to add tags/labels to an object - there the multi-selection of the combobox is really great as the added items are designed as input-chips. 
- So existing tags can be added, but of course we have the necessity **to create new tags**, that need to be added to the list.
- Datalist does not work as replacement as it does not support multi-selection.

![2023-06-23_10h49_42](https://github.com/vmware-clarity/ng-clarity/assets/311619/72a259a8-a0c7-4b1a-a277-fd017bcc0141)

## Usability guidance
Currently the differentiation when to use datalist / combobox often puzzles designers/developlers and in the end of course our users.
Having the combobox supporting custom values / free text input would be a great step to unify the user interface.

**Related discussion in core**: https://github.com/vmware-clarity/core/discussions/162

**Reply by d-m-s** [reply]
@kevinbuhmann @bbogdanov, @ashleyryan, @pdaigle,  ... anyone from vm-ware 
Can you give us any update/feedback?
